src/tokio/io.rs

```diff
@@ -28,19 +26,28 @@ where
     fn is_disconnect_error(&self, err: &io::Error) -> bool {
         use std::io::ErrorKind::*;
 
-        match err.kind() {
-            NotFound | PermissionDenied | ConnectionRefused | ConnectionReset
-            | ConnectionAborted | NotConnected | AddrInUse | AddrNotAvailable | BrokenPipe
-            | AlreadyExists => true,
-            _ => false,
-        }
+        matches!(
+            err.kind(),
+            NotFound
+                | PermissionDenied
+                | ConnectionRefused
+                | ConnectionReset
+                | ConnectionAborted
+                | NotConnected
+                | AddrInUse
+                | AddrNotAvailable
+                | BrokenPipe
+                | AlreadyExists
+        )
     }
 
     /// If the underlying IO item implements AsyncRead, this method allows the user to specify
     /// if a technically successful read actually means that the connect is closed.
     /// For example, tokio's TcpStream successfully performs a read of 0 bytes when closed.
-    fn is_final_read(&self, received_bytes: usize) -> bool {
-        received_bytes == 0 // definitely true for tcp, perhaps true for other io as well
+    fn is_final_read(&self, pre_buf_filled: usize, post_buf_filled: usize) -> bool {
+        // definitely true for tcp, perhaps true for other io as well,
+        // indicative of EOF hit
+        pre_buf_filled == post_buf_filled
```

Comment:
You are right about this not being clear in its intent. I have restored it to the previous interface. I do not want to inline the call however, since I want it to be customizable for any kind of IO, and perhaps a different IO would have some kind of other weird bytes_read behavior, and I want it to be customizable